Sources/TuistSupport/Utils/RecentPathsRecorder.swift

```diff
@@ -0,0 +1,81 @@
+import FileSystem
+import Foundation
+import Mockable
+import Path
+import ServiceContextModule
+
+private enum RecentPathsContextKey: ServiceContextKey {
+    typealias Value = RecentPathsRecording
+}
+
+extension ServiceContext {
+    public var recentPaths: RecentPathsRecording? {
+        get {
+            self[RecentPathsContextKey.self]
+        } set {
+            self[RecentPathsContextKey.self] = newValue
+        }
+    }
+}
+
+/// This is a utility to record the paths the user interacts with.
+/// This is recorded either from the working directory or the --path argument when invoking a command.
+/// The information can then be used by tools like MCP servers to allow users to interact with their most recent
+/// projects.
+@Mockable
+public protocol RecentPathsRecording {
```

Comment:
I'd personally call this as `RecentPathsStoring` rather than "recording". That also goes better with the input parameter of `storageDirectory`.